### ISSUE TITLE
node.contains(...)

### ISSUE THREAD

**Issue by mudcube** [cheton/flattree#2]
What do you think about having a node.contains(...) method? This would be very helpful, as currently it's possible to create an infinite loop when adding nodes as a sub-node to itself in InfiniteTree. 

```
contains(target) {
	const children = this.children
	const length = children.length
	for (let i = 0; i < length; i++) {
		const node = children[i]
		if (node === target) return true
		if (node.children) {
			const result = node.contains(target)
			if (result) return result
		}
	}
}
```

**Comment by cheton** [comment]
It's fine with me to add a `node.contains()` method. From your example, traversing up through its ancestors should be more efficient since the target node has a parent link:

```js
contains(target) {
    while (target && target !== this) {
        if (target.parent === this) {
            return true;
        }
        target = target.parent;
    }

    return false;
}
```

**Comment by mudcube** [comment]
Good point! That's much more efficient.

**Comment by cheton** [comment]
It's now included in 0.10.0. Thank you.
